src/main/kotlin/no/nav/k9/los/integrasjon/k9/K9SakServiceSystemClient.kt

```diff
@@ -25,28 +25,39 @@ open class K9SakServiceSystemClient constructor(
     val accessTokenClient: AccessTokenClient,
     val scope: String
 ) : IK9SakService {
-    val log = LoggerFactory.getLogger("K9SakService")!!
+    private val log = LoggerFactory.getLogger("K9SakService")!!
+
     private val cachedAccessTokenClient = CachedAccessTokenClient(accessTokenClient)
-    private val cache = Cache<Boolean>(cacheSize = 10000)
     private val url = configuration.k9Url()
     private val scopes = setOf(scope)
 
+    private val cache = Cache<UUID, Boolean>(cacheSize = 10000)
+    private val cacheObjectDuration = Duration.ofHours(12)
+
     override suspend fun refreshBehandlinger(behandlingIdList: BehandlingIdListe) {
-        // Passer på at vi ikke sender behandlingsider om igjen før det har gått 24 timer
-        val behandlingIdListe =
-            BehandlingIdListe(behandlingIdList.behandlingUuid.filter {
-                cache.setIfEmpty(it.toString(), CacheObject(true, expire = LocalDateTime.now().plusHours(12)))
-            }.map { BehandlingIdDto(it) }.take(999))
 
-        if (behandlingIdListe.behandlinger.isEmpty()) {
-            return
+        //TODO la KøOppdatert gå gjennom channel til RefreshK9 for å unngå at låsing blir nødvendig her
+        synchronized(cache) {
```

Comment:
Funker dette som i java på suspend metode?

Reply:
byttet til channel for å slippe synchronized